public/app/plugins/panel/barchart/utils.ts

```diff
@@ -302,7 +302,7 @@ function getRotationPadding(frame: DataFrame, rotateLabel: number, valueMaxLengt
   // Add padding to the bottom to avoid clipping the rotated labels.
   const paddingBottom = Math.sin(((rotateLabel >= 0 ? rotateLabel : rotateLabel * -1) * Math.PI) / 180) * maxLength;
 
-  return [0, paddingRight, paddingBottom, paddingLeft];
+  return [Math.round(UPLOT_AXIS_FONT_SIZE * uPlot.pxRatio), paddingRight, paddingBottom, paddingLeft];
```

Comment:
`uPlot.pxRatio` is a cached `window.devicePixelRatio`, because that's expensive to read directly, for some reason. :shrug:

Reply:
That makes sense, I'm guessing the browser might actually query for it when read but not 100% sure. Unrelated but this _might_ be the source of a strange font rendering error I've seen. Will follow up with you 😄